rfcs/pagination-component.md

```diff
@@ -0,0 +1,70 @@
+# Pagination Component
+
+### Exports
+
+- `Pagination`
+- `StatefulPagination`
+- `StatefulContainer`
+- `StyledRoot`
+- `StyledBackButton`
+- `StyledNextButton`
+- `StyledMaxLabel`
+- `STATE_CHANGE_TYPE`
+
+### Pagination API
+
+- `numPages: number` - Required
+  Max number of pages
+- `currentPage: number` - Required
+  The current page
+- `prepositionLabel: string` - Optional
+  Defaults to "of"
+- `overrides: {PrevButton, NextButton, Select}` - Optional
+  Presentational override props, all are `React.ComponentType<*> | {component, props, style}`
+- `onBackClick: (event) => void` - Optional
+  Callback for prev button click
```

Comment:
nit: Should we use `onPrevClick` so terminology is consistent with the default button label?

Reply:
Oh yeah there's some inconsistency here. I'll update everything to use `prev`.